hedera-mirror-rest/stateproof.js

```diff
@@ -0,0 +1,265 @@
+/*-
+ * ‌
+ * Hedera Mirror Node
+ *
+ * Copyright (C) 2019 - 2020 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const utils = require('./utils');
+const config = require('./config');
+const constants = require('./constants');
+const EntityId = require('./entityId');
+const TransactionId = require('./transactionId');
+const s3client = require('./s3client');
+const { DbError } = require('./errors/dbError');
+const { NotFoundError } = require('./errors/notFoundError');
+const { FileDownloadError } = require('./errors/fileDownloadError');
+
+/**
+ * Get the consensus_ns of the transaction. Throws exception if no such successful transaction found or multiple such
+ * transactions found.
+ * @param {TransactionId} transactionId
+ * @returns {Promise<String>} consensus_ns of the successful transaction if found
+ */
+let getSuccessfulTransactionConsensusNs = async (transactionId) => {
+  const sqlParams = [transactionId.getEntityId().getEncodedId(), transactionId.getValidStartNs()];
+  const sqlQuery = `SELECT consensus_ns
+       FROM transaction
+       WHERE payer_account_id = ?
+         AND valid_start_ns = ?
+         AND result = 22`; // only the successful transaction
+  const pgSqlQuery = utils.convertMySqlStyleQueryToPostgres(sqlQuery, sqlParams);
```

Comment:
old logic used by only accounts and balances I believe. Eventually we want to remove that.
Skip the `sqlQuery` definition and go straight to `pgSqlQuery` as below 
```suggestion
  const pgSqlQuery = select consensus_ns from transaction where payer_account_id = $1 and valid_start_ns = $2 and result = 22;
```

Reply:
thanks, will change it as suggested

Reply:
updated.